ambry-tools/src/main/java/com.github.ambry/store/ConsistencyCheckerTool.java

```diff
@@ -254,14 +280,19 @@ public ConsistencyCheckerTool(ClusterMap clusterMap, StoreKeyFactory storeKeyFac
       DumpIndexTool.IndexProcessingResults result = results.get(replica);
       for (Map.Entry<StoreKey, DumpIndexTool.Info> entry : result.getKeyToState().entrySet()) {
         StoreKey key = entry.getKey();
-        key = convertMap.get(key);
-        if (!keyReplicationStatusMap.containsKey(key)) {
-          keyReplicationStatusMap.put(key, new ReplicationStatus(replicas));
+        StoreKey newKey = convertMap.get(key);
+        boolean isDeprecated = false;
+        if (newKey == null) {
+          newKey = key;
+          isDeprecated = true;
+        }
+        if (!keyReplicationStatusMap.containsKey(newKey)) {
```

Comment:
you can collapse the following three lines into one now that we are using java 8
`ReplicationStatus status = keyReplicationStatusMap.computeIfAbsent(newKey, k -> new ReplicationStatus(replicas))`

Reply:
changed